src/lib/logstore/log_store_service.cpp

```diff
@@ -206,17 +208,17 @@ void LogStoreService::logdev_super_blk_found(const sisl::byte_view& buf, void* m
         auto id = sb->logdev_id;
         LOGDEBUGMOD(logstore, "Log dev superblk found logdev={}", id);
         const auto it = m_id_logdev_map.find(id);
-        if (it == m_id_logdev_map.end()) {
-            LOGERROR("logdev={} found but not opened yet, it will be discarded after logstore is started", id);
-            m_unopened_logdev.insert(id);
-        }
-
         // We could update the logdev map either with logdev or rollback superblks found callbacks.
         if (it != m_id_logdev_map.end()) {
             logdev = it->second;
         } else {
             logdev = std::make_shared< LogDev >(id, m_logdev_vdev.get());
             m_id_logdev_map.emplace(id, logdev);
+            // when recover logdev meta blk, we get all the logdevs from the superblk. we put them in m_unopened_logdev
+            // too. after logdev meta blks are all recovered, when a client opens a logdev, we remove it from
+            // m_unopened_logdev. so that when we start log service, all the left items in m_unopened_logdev are those
+            // not open, which can be destroyed
+            m_unopened_logdev.insert(id);
```

Comment:
You will have to add the same in rollback_super_blk_found too.

Reply:
@sanebay 
pls take a look at line 54 in this file
```
    meta_service().register_handler(
        logdev_rollback_sb_meta_name,
        [this](meta_blk* mblk, sisl::byte_view buf, size_t size) {
            rollback_super_blk_found(std::move(buf), voidptr_cast(mblk));
        },
        nullptr, true, std::optional< meta_subtype_vec_t >({logdev_sb_meta_name}));
```

this will make sure all the `rollback_super_blk_found` will be called after all the `logdev_super_blk_found` is called . order is guaranteed here.